tests/integration/blockchain/test_transcoding.py

```diff
@@ -0,0 +1,148 @@
+import logging
+import pathlib
+import time
+
+import lbry.wallet  # just to make the following line work:
+from lbry.conf import TranscodeConfig
+from lbry.file_analysis import VideoFileAnalyzer
+from .test_claim_commands import ClaimTestCase
+
+log = logging.getLogger(__name__)
+
+
+class MeasureTime:
+    def __init__(self, text):
+        print(text, end="...", flush=True)
+
+    def __enter__(self):
+        self.start = time.perf_counter()
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        end = time.perf_counter()
+        print(f" done in {end - self.start:.6f}s", flush=True)
+
+
+class TranscodeValidation(ClaimTestCase):
+
+    def make_name(self, name, extension=""):
+        path = pathlib.Path(self.video_file_name)
+        return path.parent / f"{path.stem}_{name}{extension or path.suffix}"
+
+    async def asyncSetUp(self):
+        await super().asyncSetUp()
+        self.conf = TranscodeConfig()
+        self.conf.volume_analysis_time = 0  # disable it as the test file isn't very good here
+        self.analyzer = VideoFileAnalyzer(self.conf)
+        file_ogg = self.make_name("ogg", ".ogg")
+        if not file_ogg.exists():
+            command = f'-i "{self.video_file_name}" -c:v libtheora -q:v 4 -c:a libvorbis -q:a 4 ' \
+                      f'-c:s copy -c:d copy "{file_ogg}"'
+            with MeasureTime(f"Creating {file_ogg.name}"):
+                output, code = await self.analyzer._execute("ffmpeg", command)
+                self.assertEqual(code, 0, output)
+
+        file_webm = self.make_name("webm", ".webm")
+        if not file_webm.exists():
+            command = f'-i "{self.video_file_name}" -c:v libvpx-vp9 -crf 36 -b:v 0 -cpu-used 2 ' \
+                      f'-c:a libopus -b:a 128k -c:s copy -c:d copy "{file_webm}"'
+            with MeasureTime(f"Creating {file_webm.name}"):
+                output, code = await self.analyzer._execute("ffmpeg", command)
+                self.assertEqual(code, 0, output)
+
+        self.should_work = [self.video_file_name, str(file_ogg), str(file_webm)]
+
+    async def test_should_work(self):
+        for should_work_file_name in self.should_work:
+            new_file_name = await self.analyzer.verify_or_repair(True, False, should_work_file_name)
+            self.assertEqual(should_work_file_name, new_file_name)
```

Comment:
Can you unwrap this loop, there are only 3 items in the list. We have a practice of not using loops in tests if at all possible because it makes it impossible to set a breakpoint at a failing part of the test.

Reply:
Fixed and pushed.